File: examples/frameworks/typescript/mail.ts
// The mock `mail` tool that the agents call.
import { env } from "node:process";
import { execution } from "rebuno";

const MAIL_URL = env.MAIL_URL;

export const mail = {
  /** Returns a delivery receipt, posting the message to MAIL_URL first if that is set. */
  async send(to: string, body: string) {
    if (!MAIL_URL) return { status: "sent", to, simulated: true, detail: `simulated email to ${to}; no delivery endpoint configured` };
    const resp = await fetch(MAIL_URL, {
      method: "POST",
      signal: AbortSignal.timeout(30_000),
      headers: { "content-type": "application/json", "X-Execution-Id": execution().id },
      body: JSON.stringify({ to, body }),
    });
    if (!resp.ok) throw new Error(`Mail endpoint returned HTTP ${resp.status}`);
    return resp.json();
  },
};

File: examples/frameworks/python/mail.py
"""The mock `mail` tool that the agents call."""

import os

import httpx2
from rebuno import execution

MAIL_URL = os.environ.get("MAIL_URL")


async def send(to: str, body: str) -> dict:
    """Return a delivery receipt, posting the message to MAIL_URL first if that is set."""
    if not MAIL_URL:
        return {
            "status": "sent",
            "to": to,
            "simulated": True,
            "detail": f"simulated email to {to}; no delivery endpoint configured",
        }
    async with httpx2.AsyncClient() as http:
        response = await http.post(
            MAIL_URL,
            json={"to": to, "body": body},
            headers={"X-Execution-Id": str(execution().id)},
            timeout=30,
        )
        response.raise_for_status()
        return response.json()
